src/osd/osd_types.cc

```diff
@@ -3770,11 +3770,14 @@ void pg_hit_set_info_t::encode(bufferlist& bl) const
 
 void pg_hit_set_info_t::decode(bufferlist::iterator& p)
 {
-  DECODE_START(1, p);
+  DECODE_START_LEGACY_COMPAT_LEN(2, 1, 1, p);
```

Comment:
You don't need to use DECODE_START_LEGACY_COMPAT_LEN here.  DECODE_START(1, p) will be fine.